HeterogeneousCore/AlpakaTest/plugins/TestAlpakaAnalyzer.cc

```diff
@@ -134,19 +139,32 @@ class TestAlpakaAnalyzer : public edm::global::EDAnalyzer<> {
       assert(vi.id() == i);
       assert(vi.m() == matrix * i);
     }
+
+    if (expectBackend_) {
+      auto const& backend = event.get(backendToken_);
+      if (*expectBackend_ != backend) {
+        throw cms::Exception("Assert") << "Expected input backend " << *expectBackend_ << ", got " << backend;
+      }
+    }
   }
 
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
     edm::ParameterSetDescription desc;
     desc.add<edm::InputTag>("source");
     desc.add<int>("expectSize", -1)
         ->setComment("Expected size of the input collection. Values < 0 mean the check is not performed. Default: -1");
+    desc.add<std::string>("expectBackend", "")
+        ->setComment(
+            "Expected backend of the input collection. Empty value means to not perform the check. Default: empty "
+            "string");
     descriptions.addWithDefaultLabel(desc);
   }
 
 private:
   const edm::InputTag source_;
   const edm::EDGetTokenT<portabletest::TestHostCollection> token_;
+  edm::EDGetTokenT<short> backendToken_;
+  std::optional<short> expectBackend_;
```

Comment:
why not
```c++
  std::optional<cms::alpakatools::Backend> expectBackend_;
```
?

Reply:
In order to have one `static_cast<short>` in the construction instead of two in the `analyze()`. If the printout https://github.com/cms-sw/cmssw/pull/41564#discussion_r1203982125 would be changed to string representation, then I'd change to use the `cms::alpakatools::Backend` here.

Reply:
The `Backend` enumerator is now used as the class member. I changed the corresponding printout to use the string presentation (it came to my mind that the string representations could be useful e.g. in DQM histograms).